### DISCUSSION TITLE
ansible-galaxy is not logging an error on permission denied

### DISCUSSION THREAD

**Discussion by rroemer** [renovatebot/renovate#15851] [Request Help] [answered]
Hi

When running renovate against a project with ansible roles in a private repository, then nothing is logged if permissions is denied.
I expect permission denied errors to be logged at least as Info.

Is this a bug, or by design?

```
INFO: Autodiscovered repositories
      "length": 1,
      "repositories": ["<redacted>"]
INFO: Repository started (repository=<redacted>)
      "renovateVersion": "32.71.3"
INFO: Dependency extraction complete (repository=<redacted>)
      "baseBranch": "master",
      "stats": {
        "managers": {
          "ansible-galaxy": {"fileCount": 1, "depCount": 14}
        },
        "total": {"fileCount": 1, "depCount": 14}
      }
INFO: Repository finished (repository=<redacted>)
      "durationMs": 7195
```

If debug logging is enabled we get the following output

```
DEBUG: Failed to look up dependency <redacted> (repository=<redacted>, packageFile=ansible/requirements.yml, dependency=<redacted>)
DEBUG: Datasource unknown error (repository=<redacted>)
       "datasource": "git-tags",
       "packageName": "git@gitlab.<redacted>.git",
       "err": {
         "task": {
           "commands": [
             "ls-remote",
             "git@gitlab.<redacted>.git"
           ],
           "format": "utf-8",
           "parser": "[function]"
         },
         "message": "git@gitlab.<redacted>: Permission denied (publickey).\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\n",
         "stack": "Error: git@gitlab.<redacted>: Permission denied (publickey).\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\n\n    at Object.action (/usr/src/app/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:38:28)\n    at PluginStore.exec (/usr/src/app/node_modules/simple-git/src/lib/plugins/plugin-store.ts:24:29)\n    at /usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:114:40\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:111:14)\n    at GitExecutorChain.<anonymous> (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:88:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/usr/src/app/node_modules/simple-git/dist/cjs/index.js:55:24)"
       }
```

**Comment by rarkins** [comment, marked as answer]
It is expected that it's logged at DEBUG level as you've observed

**Reply by rroemer** [reply]
But why? It makes it impossible to detect if it is not working unless we log everything in debug.

**Reply by rarkins** [reply]
These lookup errors are logged in the onboarding PR body. It's assumed that if people ignore those errors and merge the onboarding PR anyway, they are not bothered by them.

#15651 is related

**Reply by rroemer** [reply]
That is not what I see. 
This project has been updated by renovate for a long time, so guess its already onboarded?
Everything was updated when the ssh-key was revoked. Then I personally release a new version of an ansible role, and I expected renovate to create a Merge Request, but nothing happened. I had to enable debug logging in order to see what was wrong.

I fixed the issue, and renovate created the MR as expected. 

Then for the test I revoked the ssh-key again, and then renovate closed the MR with this message and updated the title with - autoclosed:
![image](https://user-images.githubusercontent.com/25124388/171736680-a5cb05d5-b933-44e9-955b-9256b906e173.png)

Have I configured something wrong?
I'm using Gitlab. Config attached.
[renovate.js.txt](https://github.com/renovatebot/renovate/files/8827837/renovate.js.txt)
